pkg/api/jobartifacts/query_test.go

```diff
@@ -10,72 +10,110 @@ import (
 
-	query := JobArtifactQuery{GcsBucket: gcsClient, ContentMatcher: NewStringMatcher("ClusterVersion:", 0, 0, maxFileMatches)}
+	query := baseTestingJAQ(t, "", NewStringMatcher("ClusterVersion:", 0, 0, maxFileMatches))
 	artifact := query.getFileContentMatches(1898704060324777984, &storage.ObjectAttrs{Name: filePath})
 	assert.Empty(t, artifact.Error)
-	assert.False(t, artifact.MatchedContent.(ContentLineMatches).Truncated, "expected no need for truncating the content matches")
-	assert.Equal(t, 2, len(artifact.MatchedContent.(ContentLineMatches).Matches), "expected content to match with two lines")
+	assert.False(t, artifact.MatchedContent.ContentLineMatches.Truncated, "expected no need for truncating the content matches")
+	assert.Equal(t, 2, len(artifact.MatchedContent.ContentLineMatches.Matches), "expected content to match with two lines")
 
 	query.ContentMatcher = NewStringMatcher("error:", 0, 0, maxFileMatches)
 	artifact = query.getFileContentMatches(1898704060324777984, &storage.ObjectAttrs{Name: filePath})
 	assert.Empty(t, artifact.Error)
-	assert.True(t, artifact.MatchedContent.(ContentLineMatches).Truncated, "expected to truncate content matches")
-	assert.Equal(t, maxFileMatches, len(artifact.MatchedContent.(ContentLineMatches).Matches), "expected content to match with many lines")
+	assert.True(t, artifact.MatchedContent.ContentLineMatches.Truncated, "expected to truncate content matches")
+	assert.Equal(t, maxFileMatches, len(artifact.MatchedContent.ContentLineMatches.Matches), "expected content to match with many lines")
 }
 
 func TestFunctional_GzipContent(t *testing.T) {
-	gcsClient := util.GetGcsBucket(t)
 	const filePath = "logs/periodic-ci-openshift-release-master-ci-4.19-e2e-aws-ovn-techpreview/1909930323508989952/artifacts/e2e-aws-ovn-techpreview/gather-extra/artifacts/nodes/ip-10-0-59-177.us-east-2.compute.internal/journal"
 
-	query := JobArtifactQuery{GcsBucket: gcsClient, ContentMatcher: NewStringMatcher("error", 0, 0, maxFileMatches)}
+	query := baseTestingJAQ(t, "", NewStringMatcher("error", 0, 0, maxFileMatches))
 	artifact := query.getFileContentMatches(1909930323508989952, &storage.ObjectAttrs{Name: filePath, ContentType: "application/gzip"})
 	assert.Empty(t, artifact.Error)
-	assert.True(t, artifact.MatchedContent.(ContentLineMatches).Truncated, "expected a lot of matches")
+	assert.True(t, artifact.MatchedContent.ContentLineMatches.Truncated, "expected a lot of matches")
 	assert.Contains(t,
-		artifact.MatchedContent.(ContentLineMatches).Matches[0].Match,
+		artifact.MatchedContent.ContentLineMatches.Matches[0].Match,
 		"localhost kernel: GPT: Use GNU Parted to correct GPT errors.",
 		"expected to scan uncompressed text")
 }
 
+func TestFunctional_RequeryCachedTimedOutJobArtifacts(t *testing.T) {
+	ctx := context.Background()
+	jobRunID := int64(1898704060324777984)
+	mgr := NewManager(ctx)
```

Comment:
Are these tests hitting gcs? If so they should move to the test/e2e/ directory and thus the e2e job.

Reply:
i'm not opposed to that, but be aware they expect production creds, connection, and data to be available during the test. so i've just been running them manually as unit tests (which are skipped during automated unit tests). although production GCS contents this far in the past shouldn't change, it feels a little careless to leave tests around that will break years in the future when it gets cleaned up...

Reply:
Tricky. If these are not running as part of the normal unit tests then that's ok for now.

Reply:
they compile and run as part of automated unit tests (so we at least get compile failures if code changes signatures), but they immediately self-skip when they find they do not have credentials. and now that i think of it, some of the lookups here are expecting jobRun data from the prod DB (to match the GCS content) and would probably fail against the local DB (unless that's using a backup from prod...)